src/nvim/spell.c

```diff
@@ -4266,14 +4259,13 @@ static void suggest_trie_walk(suginfo_T *su, langp_T *lp, char_u *fword, bool so
               l = utfc_ptr2len((char *)fword + sp->ts_fidx);
               if (fword_ends) {
                 // Copy the skipped character to preword.
-                memmove(preword + sp->ts_prewordlen,
-                        fword + sp->ts_fidx, l);
-                sp->ts_prewordlen += l;
+                memmove(preword + sp->ts_prewordlen, fword + sp->ts_fidx, (size_t)l);
+                sp->ts_prewordlen = (char_u)(sp->ts_prewordlen + l);
                 preword[sp->ts_prewordlen] = NUL;
               } else {
                 sp->ts_score -= SCORE_SPLIT - SCORE_SUBST;
               }
-              sp->ts_fidx += l;
+              sp->ts_fidx = (char_u)(sp->ts_fidx + l);
```

Comment:
This is tricky. An overflow can indeed happen here when there are too many composing characters.

Reply:
Should I add an assert or...?

Reply:
I don't think so.